plugins/repository-s3/src/main/java/org/opensearch/repositories/s3/S3Service.java

```diff
@@ -279,6 +279,8 @@ protected PasswordAuthentication getPasswordAuthentication() {
         }
 
         clientBuilder.socketTimeout(Duration.ofMillis(clientSettings.readTimeoutMillis));
+        clientBuilder.maxConnections(clientSettings.maxSyncConnections);
+        clientBuilder.connectionAcquisitionTimeout(Duration.ofMillis(clientSettings.connectionAcquisitionTimeoutMillis));
```

Comment:
Why do we want to increase `connectionAcquisitionTimeout`? Default is 10seconds. If we increase the number of connections, that would help the exhaustion issue, right?

Reply:
That should help , but I can't think rationale to keep them different for sync and async client .

Reply:
Increasing number of connections alone may not help. Connections can get occupied for a long time if enough CPU/disk IO is not available. Due to this requests in the connection queue can start timing out based on the set timeout. Ideally we shouldn't be increasing these timeouts and have a different producer-consumer based solution to control upload/download bursts but that is yet to come.